apps/careers-app/backend/routers/jobs.py

```
@@ -48,15 +48,18 @@ async def list_jobs(
     except httpx.RequestError as e:
         logger.error("Network error on GET /vacancies/basic-info: %s", e)
         raise HTTPException(status_code=504, detail="Upstream service unreachable")
+    except KeyError as e:
+        logger.error("Malformed token response from upstream on GET /vacancies/basic-info: missing %s", e)
+        raise HTTPException(status_code=502, detail="Invalid response from upstream auth service")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)